From 2e3d40d813aed667c045a2f49fc774931c158deb Mon Sep 17 00:00:00 2001
From: Zebreus <zebreus@zebre.us>
Date: Fri, 15 Aug 2025 18:06:25 +0200
Subject: [PATCH] Enable mmap for WASI

---
 absl/base/config.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/absl/base/config.h b/absl/base/config.h
index b1c52273..b247d084 100644
--- a/absl/base/config.h
+++ b/absl/base/config.h
@@ -364,6 +364,7 @@ static_assert(ABSL_INTERNAL_INLINE_NAMESPACE_STR[0] != 'h' ||
 //   AsmJS                             __asmjs__
 //   WebAssembly (Emscripten)          __EMSCRIPTEN__
 //   Fuchsia                           __Fuchsia__
+//   WebAssembly (WASI)                _WASI_EMULATED_MMAN (implies __wasi__)
 //
 // Note that since Android defines both __ANDROID__ and __linux__, one
 // may probe for either Linux or Android by simply testing for __linux__.
@@ -379,7 +380,8 @@ static_assert(ABSL_INTERNAL_INLINE_NAMESPACE_STR[0] != 'h' ||
     defined(__asmjs__) || defined(__EMSCRIPTEN__) || defined(__Fuchsia__) || \
     defined(__sun) || defined(__myriad2__) || defined(__HAIKU__) ||          \
     defined(__OpenBSD__) || defined(__NetBSD__) || defined(__QNX__) ||       \
-    defined(__VXWORKS__) || defined(__hexagon__) || defined(__XTENSA__)
+    defined(__VXWORKS__) || defined(__hexagon__) || defined(__XTENSA__) ||   \
+    defined(_WASI_EMULATED_MMAN)
 #define ABSL_HAVE_MMAP 1
 #endif
 
-- 
2.48.1

