From 59b4ab4b3888326164101e119a7b63e644df8623 Mon Sep 17 00:00:00 2001
From: Zebreus <zebreus@zebre.us>
Date: Thu, 26 Feb 2026 16:17:42 +0100
Subject: [PATCH] Patch rust dependencies for wasix compatibilty

---
 src/_bcrypt/Cargo.toml | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/_bcrypt/Cargo.toml b/src/_bcrypt/Cargo.toml
index 5e29663..7f24c93 100644
--- a/src/_bcrypt/Cargo.toml
+++ b/src/_bcrypt/Cargo.toml
@@ -13,7 +13,8 @@ bcrypt = "0.17"
 bcrypt-pbkdf = "0.10.0"
 base64 = "0.22.1"
 subtle = "2.6"
-getrandom = "0.3"
+getrandom = "=0.3.3"
+target-lexicon = "=0.13.2"
 
 [features]
 extension-module = ["pyo3/extension-module"]
@@ -26,3 +27,7 @@ crate-type = ["cdylib"]
 [profile.release]
 lto = "thin"
 overflow-checks = true
+
+[patch.crates-io]
+target-lexicon = { git = "https://github.com/wasix-org/target-lexicon.git", branch = "wasix-0.13.2" }
+getrandom = { git = "https://github.com/wasix-org/getrandom.git", branch = "wasix-0.3.3" }
-- 
2.51.0

