From 7c44f34b4f051c1419a500f862ab4e7369519897 Mon Sep 17 00:00:00 2001
From: Zebreus <zebreus@zebre.us>
Date: Wed, 27 Aug 2025 15:08:33 +0200
Subject: [PATCH] Don't require tzname when targeting WASI

---
 icu4c/source/common/putilimp.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/icu4c/source/common/putilimp.h b/icu4c/source/common/putilimp.h
index 8a8ff500811..8714f99a35b 100644
--- a/icu4c/source/common/putilimp.h
+++ b/icu4c/source/common/putilimp.h
@@ -147,6 +147,8 @@ typedef size_t uintptr_t;
    /* not defined */
 #elif U_PLATFORM == U_PF_HAIKU
     /* not defined, (well it is but a loop back to icu) */
+#elif defined(__wasi__)
+    /* not defined, because WASIX does not support tzname currently */
 #else
 #   define U_TZNAME tzname
 #endif
-- 
2.48.1

