From 340697a85556a8984692056c9c5afb2e174223a9 Mon Sep 17 00:00:00 2001
From: "M.Amin Rayej" <m.amin.rayej@gmail.com>
Date: Tue, 2 Jul 2024 23:26:19 +0330
Subject: [PATCH] add wasix support

---
 rltty.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rltty.c b/rltty.c
index 882a3d4..83062e1 100644
--- a/rltty.c
+++ b/rltty.c
@@ -83,7 +83,7 @@ static void
 set_winsize (tty)
      int tty;
 {
-#if defined (TIOCGWINSZ)
+#if defined (TIOCGWINSZ) && !defined(__wasm32)
   struct winsize w;
 
   if (ioctl (tty, TIOCGWINSZ, &w) == 0)
-- 
2.48.1

